apps/client/src/Components/YoutubePlayer/index.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import YouTube from 'react-youtube';
+
+export const YoutubePlayer = ({...props}) => {
```

Comment:
o componente do player do youtube tem alguns problemas. o principal é que ele não define uma interface de tipo para que quem for usar o componente possa passar os parâmetros corretos. eu refatorei isso (e o objeto de opções que é criado dentro do componente) pra ficar de exemplo:

(os números 400 e 1080 que eu coloquei como tamanho padrão foram arbitrários e aleatórios. eu não chequei pra ver se são números bons)

````typescript
import React, { FC } from 'react'
import YouTube from 'react-youtube'

export interface YoutubePlayerProps {
  height?: number,
  width?: number,
  href: string,
}

export const YoutubePlayer: FC<YoutubePlayerProps> = ({
  height = 400,
  width = 1080,
  href,
}) => {
  let position, id;
  const options = {
    width: `${width}`,
    height: `${height}`,
    playerVars: {},
  }

  if (href.substring(7,15) === "youtu.be") {
    position = href.lastIndexOf('be/')
    id = href.substr(position+3)
  } else if (href.substring(16,21) === "embed") {
    position = href.lastIndexOf('embed/')
    id = href.substr(position+6)
  } else {
    position = href.lastIndexOf('watch?v=')  
    id = href.substr(position+8)
  }

  return (
    <YouTube
      videoId={id}
      opts={options}
    />
  )
}

```